src/app/track/track.component.ts

```diff
@@ -1,346 +1,476 @@
+
+                        if (dataPoint.d2) {
+                            trackPointsUntaggedD2.push(
+                                new Feature({
+                                    distance: dataPoint.d2,
+                                    geometry: new Point(p)
+                                })
+                            );
+                        }
+                    }
+                }
+            }
+
+            //Simplify to 1 point per 2 meter
+            let lineString = new LineString(points);
+
+            let simplifiedLineStringGeometry = lineString
+                .simplify(2);
+
+            trackLines.push(
+                new Feature(
+                    simplifiedLineStringGeometry
+                )
+            );
+        }
+
+        const trackVectorSource = new VectorSource({
+            features: trackLines
         })
 
-      });
-
-    this.layers.push(trackLayer);
-    const oldColor = 'rgba(255,0,0,0.61)';
-    const newColor = 'rgba(0,255,0,0.61)';
-    const style: LiteralStyle = {
-      variables: {
-        minYear: 1850,
-        maxYear: 2015
-      },
-      symbol: {
-        symbolType: SymbolType.CIRCLE,
-        size: ['*',
-          ['interpolate', ['linear'], ['get', 'd1'], 255, 8, 0, 26],
-          1.0],
-        color: ['interpolate',
-          ['linear'],
-          ['get', 'd2'],
-          255, newColor,
-          0, oldColor
-        ],
-        opacity: 0.5
-      }
-    };
-    const pointLayer =
-      new WebGLPointsLayer(
-        {
-          visible: true,
-
-          style: style,
-          source: new VectorSource({
-            features: trackPoints
-          }),
+        const trackLayer =
+            new VectorLayer({
+                visible: true,
+                updateWhileAnimating: false,
+                updateWhileInteracting: false,
+                source: trackVectorSource,
+                style: new Style({
+                    stroke: new Stroke({
+                        width: 3,
+                        color: 'rgb(30,144,255)'
+                    })
+                })
+
+            });
+
+
+        const pointLayerTaqgedD1 =
+            new VectorLayer(
+                {
+                    title: "Left",
+                    visible: true,
+                    style: pointStyleFunction,
+                    source: new VectorSource({
+                        features: trackPointsD1
+                    }),
+
+                } as BaseLayerOptions);
+
+
+        const pointLayerTaggedD2 =
+            new VectorLayer(
+                {
+                    startActive: false,
+                    title: "Right",
+                    visible: false,
+                    style: pointStyleFunction,
+                    source: new VectorSource({
+                        features: trackPointsD2
+                    }),
+
+                } as BaseLayerOptions);
+
+
+        const pointLayerUntaggedD1 =
+            new VectorLayer(
```

Comment:
We could avoid code repetition here if we create a function that creates a VectorLayer with only the things as argument that are different, e.g. what goes into `source`, `title`, ...
This would adhere more closely to [Don't Repeat Yourself](https://de.wikipedia.org/wiki/Don’t_repeat_yourself)